tests/common/dualtor/dual_tor_io.py

```diff
@@ -0,0 +1,424 @@
+import datetime
+import threading
+import time
+import socket
+import random
+import struct
+import ipaddress
+import logging
+from netaddr import IPNetwork
+from collections import defaultdict
+
+import scapy.all as scapyall
+import ptf.testutils as testutils
+from tests.ptf_runner import ptf_runner
+
+DOWNSTREAM_DST_IP = "192.168.79.222"
+UPSTREAM_DST_IP = "192.168.128.1"
+SOCKET_RECV_BUFFER_SIZE = 10 * 1024 * 1024
+PTFRUNNER_QLEN = 1000
+VLAN_INDEX = 0
+VLAN_HOSTS = 100
+VLAN_BASE_MAC_PATTERN = "72060001{:04}"
+LAG_BASE_MAC_PATTERN = '5c010203{:04}'
+
+logger = logging.getLogger(__name__)
+
+
+class DualTorIO:
+    def __init__(self, duthost, ptfhost, ptfadapter, tbinfo, server_port, tor_port, expect_tunnel_packet, delay, timeout, io_ready):
+        self.tor_port = tor_port
+        self.server_port = server_port
+        self.duthost = duthost
+        self.ptfadapter = ptfadapter
+        self.ptfhost = ptfhost
+        self.tbinfo = tbinfo
+        self.io_ready_event = io_ready
+        self.dut_mac = self.duthost.facts["router_mac"]
+
+        self.time_to_listen = 180.0
+        self.sniff_time_incr = 60
+        self.send_interval = 0.0035 # Inter-packet interval
+        self.packets_to_send = min(int(self.time_to_listen / (self.send_interval + 0.0015)), 45000) # How many packets to be sent in send_in_background method
+
+        self.dataplane = self.ptfadapter.dataplane
+        self.dataplane.flush()
+        self.total_disrupt_time = None
+        self.disrupts_count = None
+        self.total_disrupt_packets = None
+        # This list will contain all unique Payload ID, to filter out received floods.
+        self.unique_id = list()
```

Comment:
What's the eventual expected size of this? Would we see any meaningful performance improvements from using a `set` instead?

Reply:
In the tests performed so far, this ranges between 50k to 70k.

Reply:
Have you seen any performance issues?

Reply:
Nothing measurable in seconds for this range. But, this is a useful change, should we ever want the test to run for longer duration. The `in` check with set is significantly faster.
Added a new commit to change the type from `list` to `set`.